File: app/src/main/java/com/eslamdev/islamic/view/DuaActivity.java
package com.eslamdev.islamic.view;

import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.eslamdev.islamic.R;
import com.eslamdev.islamic.adapter.DuaAdapter;
import com.eslamdev.islamic.model.Dua;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;

public class DuaActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private DuaAdapter adapter;
    private List<Object> itemsList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dua);

        recyclerView = findViewById(R.id.dua_rv);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        adapter = new DuaAdapter();
        recyclerView.setAdapter(adapter);

        loadDuaData();
    }

    private void loadDuaData() {
        try {
            InputStream is = getAssets().open("duas_collection.json");
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            String json = new String(buffer, StandardCharsets.UTF_8);
            JSONArray categoriesArray = new JSONArray(json);

            for (int i = 0; i < categoriesArray.length(); i++) {
                JSONObject categoryObj = categoriesArray.getJSONObject(i);
                String categoryTitle = categoryObj.getString("category");
                itemsList.add(categoryTitle); // إضافة عنوان الفئة

                JSONArray duasArray = categoryObj.getJSONArray("duas");
                for (int j = 0; j < duasArray.length(); j++) {
                    JSONObject duaObj = duasArray.getJSONObject(j);
                    String title = duaObj.getString("title");
                    String duaText = duaObj.getString("dua");
                    itemsList.add(new Dua(title, duaText, categoryTitle)); // إضافة الدعاء
                }
            }

            adapter.setData(itemsList);

        } catch (IOException | JSONException e) {
            e.printStackTrace();
            Toast.makeText(this, "خطأ في تحميل ملف الأدعية!", Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/com/eslamdev/islamic/presentation/ui/activity/DuaActivity.kt
package com.eslamdev.islamic.presentation.ui.activity

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.eslamdev.islamic.R
import com.eslamdev.islamic.data.repository.DuaRepository
import com.eslamdev.islamic.presentation.ui.adapter.DuaAdapter
import com.eslamdev.islamic.presentation.viewmodel.DuaViewModel
import com.eslamdev.islamic.presentation.viewmodel.DuaViewModelFactory
import kotlinx.coroutines.flow.collectLatest

class DuaActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var viewModel: DuaViewModel
    private lateinit var adapter: DuaAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_dua)

        recyclerView = findViewById(R.id.dua_rv)
        recyclerView.layoutManager = LinearLayoutManager(this)

        adapter = DuaAdapter { selectedDuaContent ->
            showContentDialog("دعاء", selectedDuaContent, R.drawable.doaa)
        }

        recyclerView.adapter = adapter

        val repository = DuaRepository(this)
        val factory = DuaViewModelFactory(repository)
        viewModel = ViewModelProvider(this, factory)[DuaViewModel::class.java]

        lifecycleScope.launchWhenStarted {
            viewModel.duaList.collectLatest { list ->
                adapter.submitList(list)
            }
        }

        viewModel.loadDuas()

        findViewById<View>(R.id.btn_back).setOnClickListener {
            finish()
        }
    }

    private fun showContentDialog(title: String, content: String, iconRes: Int) {
        val dialogView = LayoutInflater.from(this).inflate(R.layout.dialog_daily_content, null)

        val dialogTitle = dialogView.findViewById<TextView>(R.id.dialog_title)
        val dialogText = dialogView.findViewById<TextView>(R.id.dialog_text)
        val dialogIcon = dialogView.findViewById<ImageView>(R.id.dialog_icon)
        val closeButton = dialogView.findViewById<Button>(R.id.dialog_close_button)
        val copyButton = dialogView.findViewById<Button>(R.id.btn_copy)

        dialogTitle.text = title
        dialogText.text = content
        dialogIcon.setImageResource(iconRes)

        val builder = AlertDialog.Builder(this)
        builder.setView(dialogView)
        val dialog = builder.create()
        dialog.window?.setBackgroundDrawableResource(android.R.color.transparent)

        closeButton.setOnClickListener { dialog.dismiss() }

        copyButton.setOnClickListener {
            val clipboard = getSystemService(Context.CLIPBOARD_SERVICE) as android.content.ClipboardManager
            val clip = android.content.ClipData.newPlainText("Islamic App Dua", content)
            clipboard.setPrimaryClip(clip)
            Toast.makeText(this, "تم نسخ الدعاء", Toast.LENGTH_SHORT).show()
        }

        dialog.show()
    }
}